File: app/src/main/java/io/imrekaszab/algorithms/model/BubbleSort.java
package io.imrekaszab.algorithms.model;

import androidx.annotation.StringRes;

import io.imrekaszab.algorithms.R;

public class BubbleSort extends SortAlgorithm {

    @Override
    public AlgorithmType getType() {
        return AlgorithmType.SORT;
    }

    @Override
    @StringRes
    public int getTitleResId() {
        return R.string.algorithm_list_item_bubble;
    }
}

File: app/src/main/java/io/imrekaszab/algorithms/model/BubbleSort.kt
package io.imrekaszab.algorithms.model

import androidx.annotation.StringRes
import io.imrekaszab.algorithms.R

class BubbleSort : SortAlgorithm() {
    override val type = AlgorithmType.SORT

    @get:StringRes
    override val titleResId = R.string.algorithm_list_item_bubble
}
